#! /bin/sh /usr/share/dpatch/dpatch-run
## 05-mpc-honour-libout.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make MPC honour `libout', `dllout' and `install' keywords. 
## DP: See http://deuce.doc.wustl.edu/bugzilla/show_bug.cgi?id=1915.

@DPATCH@

Index: ACE_wrappers/bin/MakeProjectCreator/templates/gnu.mpd
===================================================================
--- ACE_wrappers.orig/bin/MakeProjectCreator/templates/gnu.mpd	1970-06-23 06:23:54.000000000 +0100
+++ ACE_wrappers/bin/MakeProjectCreator/templates/gnu.mpd	2004-10-26 13:24:04.203972744 +0200
@@ -4,7 +4,7 @@
 #
 # @file <%project_file%>
 #
-# gnu.mpd,v 1.95 2004/07/01 19:00:58 elliott_c Exp
+# gnu.mpd,v 1.101 2004/09/27 19:05:33 elliott_c Exp
 #
 # This file was automatically generated by MPC.  Any changes made directly to
 # this file will be lost the next time it is generated.
@@ -87,6 +87,30 @@
 TAO_IDLFLAGS = <%tao_idlflags%>
 <%endif%>
 
+<%if(exename)%>
+<%if(install)%>
+INSBIN ?= <%install%>
+<%endif%>
+ifeq ($(INSBIN),.)
+  ifeq ($(PWD),)
+    PWD=$(shell pwd)
+  endif
+  INSBIN = $(PWD)
+endif
+OUTPUT_DIRECTORY = $(INSBIN)
+<%else%>
+<%if(dllout || libout)%>
+INSLIB ?= <%if(dllout)%><%dllout%><%else%><%libout%><%endif%>
+<%endif%>
+ifeq ($(INSLIB),.)
+  ifeq ($(PWD),)
+    PWD=$(shell pwd)
+  endif
+  INSLIB = $(PWD)
+endif
+OUTPUT_DIRECTORY = $(INSLIB)
+<%endif%>
+
 include $(ACE_ROOT)/include/makeinclude/wrapper_macros.GNU
 ## We don't need the ACELIB setting from wrapper_macros.GNU
 ACELIB =
@@ -105,23 +129,12 @@
 
 endif
 
-<%if(exename)%>
-INSBIN ?= <%if(install)%><%install%><%else%>.<%endif%>
-OUTPUT_DIRECTORY = $(INSBIN)
-<%else%>
-INSLIB ?= <%if(dllout)%><%dllout%><%else%><%libout%><%endif%>
-ifeq ($(INSLIB),.)
-  ifeq ($(PWD),)
-    PWD=$(shell pwd)
-  endif
-  INSLIB = $(PWD)
-endif
-OUTPUT_DIRECTORY = $(INSLIB)
-<%endif%>
+ifneq ($(OUTPUT_DIRECTORY),)
 all: $(OUTPUT_DIRECTORY)
 
 $(OUTPUT_DIRECTORY):
 	-@mkdir -p "$(OUTPUT_DIRECTORY)"
+endif
 
 <%if(sharedname || staticname || exename)%>
 <%if(libs && libpaths)%>
@@ -287,12 +300,14 @@
 include $(ACE_ROOT)/include/makeinclude/macros.GNU
 include $(ACE_ROOT)/include/makeinclude/rules.common.GNU
 include $(ACE_ROOT)/include/makeinclude/rules.nonested.GNU
-<%if(install && exename)%>
+<%if(exename)%>
 
+ifneq ($(OUTPUT_DIRECTORY),)
 ifneq ($(OUTPUT_DIRECTORY),.)
   INSTALL = $(VBIN:%=$(INSBIN)/%$(EXEEXT))
   CLEANUP_INSTALL += $(CLEANUP_BIN:%=$(INSBIN)/%$(EXEEXT))
 endif
+endif
 
 <%endif%>
 <%if(sharedname || staticname)%>

